envbuilder.go

```diff
@@ -156,6 +157,34 @@ func Run(ctx context.Context, options Options) error {
 		}
 	}
 
+	gitURLParsed, err := url.Parse(options.GitURL)
+	if err != nil {
+		return fmt.Errorf("invalid git URL: %w", err)
+	}
+	// If we're cloning over SSH, we need a known_hosts file.
+	if gitURLParsed.Scheme == "ssh" {
+		var knownHostsContent []byte
+		if options.GitSSHKnownHostsBase64 != "" {
+			if kh, err := base64.StdEncoding.DecodeString(options.GitSSHKnownHostsBase64); err != nil {
+				return fmt.Errorf("invalid known_hosts content: %w", err)
+			} else {
+				knownHostsContent = kh
+			}
+		} else {
+			kh, err := GenerateKnownHosts(options.Logger, gitURLParsed)
+			if err != nil {
+				return fmt.Errorf("invalid known_hosts content: %w", err)
+			} else {
+				knownHostsContent = kh
+			}
+		}
+		knownHostsPath := filepath.Join(MagicDir, "known_hosts")
+		if err := os.WriteFile(knownHostsPath, knownHostsContent, 0644); err != nil {
```

Comment:
Why do we need a file if we set the env?

Reply:
The default known hosts callback will read the file. I could possibly create a custom known hosts callback that simply iterates over the content of the known_hosts file. WDYT?

Reply:
If it's fairly trivial, go for it 👍🏻. If not, let's at least properly document why we do it.

Reply:
Even more trivial is simply not doing it!